atlp-rwanda/stackup2-barefoot-backend
# #172512095 Fix the verify endpoint

#### What does this PR do?
- Fix the verify endpoint in the following cases:
  - When the token is missing or invalid
  - When the user has already verified their account

#### Description of Task to be completed?
- Write integration tests for the verify endpoint
- Return an appropriate error message when the token is missing or invalid
- Return an appropriate error message when the user has already verified their account
- Refactor tests to integrate new changes

#### How should this be manually tested?
- Clone the project repo
- Run `cd stackup2-barefoot-backend`
- Rename `.env.example` to `.env` and provide all the required values
- Run the following commands respectively:
> `git fetch origin bg-fix-verify-endpoint:bg-fix-verify-endpoint`
> `git checkout bg-fix-verify-endpoint`
> `npm install`
> `npm run create-db`
> `npm run migrations`
> `npm run dev`

- To verify a user after signup, send the following request in POSTMAN:
  `GET http://<your_server_hostname>:<server_port>/api/auth/verify?token={{token}}`

- Send the same request again
- Remove the token from the query partially or all of it and send the request again

#### Any background context you want to provide?
  You need to sign up first to receive a token you will use on verify endpoint

#### What are the relevant pivotal tracker stories?
  [#172512095](https://www.pivotaltracker.com/story/show/172512095)

#### Screenshots (if appropriate)
![Screenshot from 2020-05-02 03-49-48](https://user-images.githubusercontent.com/45935765/80852403-85d30100-8c28-11ea-9fb3-302da8348a7b.png)

![Screenshot from 2020-05-02 03-47-56](https://user-images.githubusercontent.com/45935765/80852404-85d30100-8c28-11ea-936c-be7f3a1710e5.png)

![Screenshot from 2020-05-02 03-50-21](https://user-images.githubusercontent.com/45935765/80852416-97b4a400-8c28-11ea-9183-11d2454a6c69.png)
